.travis.yml

```diff
@@ -0,0 +1,93 @@
+language: go
+go:
+  - 1.16
+python:
+  - 3.4
+
+before_install:
+- |
+    git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '\.md$' || {
+      echo "Only docs were updated, stopping build process."
+      exit
+    }
+
+    
+jobs:
+  include:
+  - name: "Check Header"
+    env: TEST=check_header
+  - name: "Build"
+    env: TEST=build
+  - name: "Unit Test"
+    env: TEST=unit_test
+  - name: "Code Scan"
+    env: TEST=code_scan
+  - name: "Smoke Test"
+    env: TEST=smoke_test
+ 
+script: >
+  if [ x${TEST} = xcheck_header ]; then
+    git clone --branch=devops https://github.com/vmware/singleton.git devops
+    cp $TRAVIS_BUILD_DIR/devops/check_headers.py .
+    chmod +x check_headers.py
+    python3 ./check_headers.py -f "$(git diff --name-only --diff-filter=d $TRAVIS_COMMIT_RANGE)"
+  fi &&
+
+  if [ x${TEST} = xbuild ]; then
+    make build
+  fi &&
+
+  if [ x${TEST} = xunit_test ]; then
+    make test
+  fi &&
+
+  if [ x${TEST} = xcode_scan ]; then
+    SONARSCANNER='sonar-scanner-cli-4.0.0.1744-linux'
+    SONARSCANNERINNER='sonar-scanner-4.0.0.1744-linux'
+    wget https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/$SONARSCANNER.zip
+    unzip $SONARSCANNER.zip
+    rm -f $SONARSCANNER.zip
+    make test -json > report.json
+    make coverage 
+    git clone --branch=devops https://github.com/vmware/singleton.git devops
+    str=$(printf '=%.0s' {1..50})
+    echo $str Unit Test Start $str
```

Comment:
I see below messages in log. Please check.

str=$(printf '=%.0s' {1..50})
  echo $str Unit Test Start $str